wpical/src/main/native/cpp/WPIcal.cpp

```diff
@@ -58,6 +60,81 @@ void drawCheck() {
   ImGui::NewLine();
 }
 
+void openFileButton(const char* text, std::string& selected_file,
+                    const std::string& file_type,
+                    const std::string& file_extensions) {
+  if (ImGui::Button(text)) {
+    auto selector = std::make_unique<pfd::open_file>(
+        "Select File", "", std::vector<std::string>{file_type, file_extensions},
+        pfd::opt::none);
+
+    if (selector) {
```

Comment:
This doesn't work.  The if (selector) piece needs to be continuously called (e.g. from DisplayGui()); here it's only being called if the button is being pressed.  You need to save the selector to a static variable and always call the if (selector) piece.  The previous version of the code did it this way within DisplayGui().